core/src/main/java/org/apache/brooklyn/core/mgmt/classloading/JavaBrooklynClassLoadingContext.java

```diff
@@ -82,6 +87,10 @@ private ClassLoader getClassLoader() {
         try {
             className = DeserializingClassRenamesProvider.findMappedName(className);
             return (Maybe) Maybe.of(getClassLoader().loadClass(className));
+        } catch (NoClassDefFoundError e) {
+            String msg = "Invalid linkage in (transitive dependencies of) class "+className+": "+e.toString();
+            LOG.debug(msg);
```

Comment:
I'll live with flooding the debug log, as folk should really fix he underlying problem.
Longer term, we could guard to only log.warn once about it for a given class.